simple-backend/nlpviewer_backend/lib/stave_project.py

```diff
@@ -0,0 +1,254 @@
+                errno.ENOENT, os.strerror(errno.ENOENT), self._project_path)
+
+        # Load project meta data
+        with open(os.path.join(
+            self._project_path, self.PROJECT_META_FILE), "r") as f:
+            self._project_meta: Dict = json.load(f)
+
+    @property
+    def ontology(self):
+        return self._project_meta.get(self.ONTO_FIELD)
+
+    @property
+    def project_configs(self):
+        return self._project_meta.get(self.CONF_FIELD)
+
+    @property
+    def project_name(self):
+        return self._project_meta.get(self.NAME_FIELD)
+    
+    @property
+    def project_type(self):
+        return self._project_meta.get(self.TYPE_FIELD)
+    
+    @property
+    def multi_ontology(self):
+        return self._project_meta.get(self.MULT_FIELD)
+    
+    def get_textpack(self, index: int):
+        """
+        Get the target textpack based on input index. Will raise an exception
+        if the index is invalid or not found.
+
+        Args:
+            index: The index of target textpack.
+
+        Returns:
+            textpack: A dictionary representing a textpack.
+        """
+        with open(self.get_textpack_file(index), "r") as f:
+            textpack = json.load(f)
+        return textpack
+
+    def get_textpack_prefix(self, index: int):
+        textpack_file = os.path.basename(self.get_textpack_file(index))
+        return textpack_file[:-len(f".{index}.json")]
+
+    def get_textpack_file(self, index: int):
+        # TODO: The following implementation uses glob to find target 
+        #       textpack. It is easy for maintenance but can be less
+        #       efficient in some use cases. May find better way to track
+        #       files or may add a layer of caching in future version.
+        file_pattern = os.path.join(self._project_path, f"*.{index}.json")
+        match = glob.glob(file_pattern)
+        if len(match) != 1:
+            if len(match) == 0:
+                raise FileNotFoundError(
+                    errno.ENOENT, os.strerror(errno.ENOENT), file_pattern)
+            raise Exception("Duplicate textpack index.")
+        return match[0]
+
+    def get_next_index(self, index: int):
+        match = glob.glob(
+            os.path.join(self._project_path, f"*.{index + 1}.json"))
+        return index if len(match) == 0 else (index + 1)
+    
+    def get_prev_index(self, index: int):
+        return max(index - 1, 0)
+
+
+class StaveProjectWriter(StaveProject):
+    """
+    Serialize a Stave project onto local disk storage. It will follow the same
+    directory specification as StaveProjectReader:
+
+        - project_path/
+            - PROJECT_META_FILE
+            - *.0.json
+            - *.1.json
+            - *.2.json
+            ...
+
+    Any Stave project can be flushed to disk in the format above.
+    Example usage:
+
+        project_writer = StaveProjectWriter(
+            project_path=project_path,
+            project_name=project_name,
```

Comment:
I think given a unique project (say given the project id), we should be able to find all the information of the project and serialize them. Do you have another script to do this instead?

Reply:
I originally planned to create another issue&PR for import/export scripts after this PR is merged. Or maybe it's better to put them in the same PR?